packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts

```diff
@@ -109,6 +108,13 @@ export class CoreVersionedRoute implements VersionedRoute {
         body: `No handlers registered for [${this.method}] [${this.path}].`,
       });
     }
+
+    if (!this.hasVersion(req) && (this.isInternal || this.router.isDev)) {
```

Comment:
This is the most important change that everything else pivots on.

Reply:
Have we validated with solutions if they are willing to specify the version for every request?

Could this lead to making internal endpoints public because those don't have the same limitation? #lazyDev

Reply:
Haha, that is not something I had thought of... However, if we are in dev we require headers for public and internal endpoints.